### PR TITLE
Create a singleLineTextField component and singleLineTextField option editor

### PR DESCRIPTION
**Pull request by niuyunsi**

### CODE REVIEW

File: src/components/Fields/SingleLineTextField.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import * as React from 'react';
+import { IField } from '../../data'
+
+interface IState {
+}
+
+interface IProps {
+    field: IField;
+    value: string,
```
Comment (linweigao) [inline src/components/Fields/SingleLineTextField.tsx]: value?

File: src/components/Fields/SingleLineTextField.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    onChange: (newValue: string) => void;
```
Comment (linweigao) [inline src/components/Fields/SingleLineTextField.tsx]: onChange?

File: src/components/Fields/SingleLineTextField.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+export default class SingleLineTextField extends React.PureComponent<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+        this.onChange = this.onChange.bind(this);
+    }
+
+    public render() {
+        const { label, hint, required, allowMutiple } = this.props.field.options ? this.props.field.options : null;
```
Comment (linweigao) [inline src/components/Fields/SingleLineTextField.tsx]: allowMutiple => unique

File: src/components/Fields/SingleLineTextField.tsx (hunk 1 above)
Comment (niuyunsi) [reply src/components/Fields/SingleLineTextField.tsx]: changed to textFieldValue

File: src/components/Fields/SingleLineTextField.tsx (hunk 2 above)
Comment (niuyunsi) [reply src/components/Fields/SingleLineTextField.tsx]: changed to textFieldOnChange

File: src/components/Fields/SingleLineTextField.tsx (hunk 3 above)
Comment (niuyunsi) [reply src/components/Fields/SingleLineTextField.tsx]: done

File: src/components/Fields/SingleLineTextField.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+}
+
+export default class SingleLineTextField extends React.PureComponent<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+        this.textFieldOnChange = this.textFieldOnChange.bind(this);
+    }
+
+    public render() {
+        const { label, hint, required, unique } = this.props.field.options ? this.props.field.options : null;
```
Comment (linweigao) [inline src/components/Fields/SingleLineTextField.tsx]: unique is not used. You can remove it.

**Review by linweigao — APPROVED** [review]

File: src/components/Fields/SingleLineTextField.tsx (hunk 4 above)
Comment (niuyunsi) [reply src/components/Fields/SingleLineTextField.tsx]: done

**Pull request merged**